R/lint.R

```diff
@@ -376,6 +378,23 @@ find_package <- function(path) {
   path
 }
 
+has_rproj <- function(path) {
+  length(list.files(path = path, pattern = "\\.Rproj$")) > 0L
+}
+
+find_rproj <- function(path) {
+  path <- normalizePath(path, mustWork = FALSE)
+
+  while (!has_rproj(path)) {
+    path <- dirname(path)
+    if (is_root(path)) {
+      return(NULL)
+    }
+  }
+
+  list.files(path = path, pattern = "\\.Rproj$", full.names = TRUE)[1L]
```

Comment:
Nit: this runs an extra `list.files()` right? would it work for the `length()` check to happen in the `while()` loop instead?

Reply:
I could refactor `has_rproj()` to `find_rproj_at()` returning the file name if found, turning it into `while(is.null(res <- find_rproj_at(path)) { ... }`.
WDYT?

Reply:
is.null() or length()? either way sounds like the right idea

Reply:
Yeah, moving length() outside and using `head(character(), 1L)` inside seems like the most elegant solution. Thanks for your suggestion.